assets/css/src/amp-default.css

```diff
@@ -1,13 +1,10 @@
-
 /*
- * Prevent cases of amp-img converted from img to appear with stretching by using object-fit to scale.
+ * Prevent cases of amp-img converted from img to appear with stretching by using object-fit to scale. Only do this for
+ * amp-img/amp-anim elements that we converted from <img> via AMP_Img_Sanitizer. This is key for images that get 100%
+ * width such as in the post content so that the contents do not get stretched or cropped.
  * See <https://github.com/ampproject/amphtml/issues/21371#issuecomment-475443219>.
- * Also use object-fit:contain in worst case scenario when we can't figure out dimensions for an image.
- * Additionally, in side of \AMP_Img_Sanitizer::determine_dimensions() it could $amp_img->setAttribute( 'object-fit', 'contain' )
- * so that the following rules wouldn't be needed.
  */
-amp-img.amp-wp-enforced-sizes[layout="intrinsic"] > img,
-amp-anim.amp-wp-enforced-sizes[layout="intrinsic"] > img {
+.amp-wp-enforced-sizes {
```

Comment:
There is a chance that this may have too much specificity and override user styles, although this is very unlikely. 

There is a new hack to implement low specificity selector as seen here:

https://github.com/ampproject/amphtml/blob/ec3090a369633ea930ac8d85e211c8bd2451cd53/extensions/amp-accordion/1.0/amp-accordion.css#L41-L53

Example: https://codepen.io/westonruter/pen/JjENYGK

This being the case, this selector could actually be changed to:

```suggestion
:where(.amp-wp-enforced-sizes) {
```

And this would ensure that if anyone has a bare `img { object-fit:cover }` it won't get overridden by the default style here.

Nevertheless, `:where` is not supported yet by all browsers: https://caniuse.com/mdn-css_selectors_where

It is very unlikely that someone is going to want to style all images with `object-fit`, and the examples of using `object-fit` in core all make use of selectors that have much more specificity than [0.1.0](https://polypane.app/css-specificity-calculator/#selector=.amp-wp-enforced-sizes):

Selector | Specificity
---------|-----------
`.has-header-image .custom-header-media img` | 0.2.1
`.has-header-video .custom-header-media video` | 0.2.1
`.has-header-image:not(.twentyseventeen-front-page):not(.home) .custom-header-media img` | 0.4.1
`.site-featured-image .post-thumbnail img` | 0.2.1
`.site-header.featured-image .site-featured-image .post-thumbnail img` | 0.4.1

Reply:
Although unlikely, I could see someone using something like `header > img { object-fit:cover }`, which would have a specificity of [0.0.2](https://polypane.app/css-specificity-calculator/#selector=header%20%3E%20img).

Unfortunately `@supports selector` is [not widely supported](https://caniuse.com/mdn-css_at-rules_supports_selector) by all browsers either so we cannot rely on that.